### ISSUE TITLE
elasticinbox and CQL3 / Cassandra 2

### ISSUE THREAD

**Issue by johnwiesel** [elasticinbox/elasticinbox-java#51]
Hello,

can you elaborate a bit on elasticinbox and integration with cql3 / Cassandra 2?

We recently migrated our applications to datastax' java driver and dropped astyanax and hector, and thinking of switching to Cassandra 2 soon, that's why I am asking.

Thank you and best wishes,
John

**Comment by johnwiesel** [comment]
This is related to #41

**Comment by rstml** [comment]
ElasticInbox currently uses Thrift. Thrift is not going anywhere and supported on C\* 2.x.

Going forward I was thinking that Java binary driver would be the best option. However, CQL3 support may require some breaking changes. #41 is opened to evaluate it.

CQL3 has some great features and one of them is coming in 2.1: https://issues.apache.org/jira/browse/CASSANDRA-4511 With secondary indexes on collections we can stop creating reverse index for labels in ElasticInbox.

**Comment by johnwiesel** [comment]
I agree that this would be an improvement. And I also agree on the choice of driver. (Well the options seem to be limited anyway.)

Afair the original plan for the C\* 2.1 release was January 2014, but it looks like there is still a bit of work needed before a release. Also, it might make sense to wait for first bugfix releases for the 2.1 version before switching, right? So EI+2.1 will probably be several months away at least.

Looking into a switch to CQL3 pre-2.1 might ease the burden of migrating later. Do you have some insight into where most of the work in EI would be necessary?
